runtime/gc_realtime/MetronomeAlarm.cpp

```diff
@@ -207,29 +207,22 @@ MM_ITAlarm::initialize(MM_EnvironmentBase *env, MM_MetronomeAlarmThread* alarmTh
 	}
 	
 #if defined(WIN32)
-	TIMECAPS tc;
-	if (timeGetDevCaps(&tc, sizeof(TIMECAPS)) != TIMERR_NOERROR) {
-		return false;
-	}
-	UINT uDesiredPeriod = (UINT)(_extensions->itPeriodMicro / 1000);
-	UINT uPeriod = (uDesiredPeriod < tc.wPeriodMin) ? tc.wPeriodMin : uDesiredPeriod;
-	UINT uDelay = uPeriod;
-	if (timeBeginPeriod(uPeriod) != TIMERR_NOERROR) {
-		/* Error beginning higher frequency period  */
-		return false;
-	}
-	_uTimerId = (UINT)timeSetEvent(uDelay, uPeriod, itAlarmThunk, (DWORD_PTR)alarmThread, TIME_PERIODIC);
-	if (_uTimerId == 0) {
+	OMRPORT_ACCESS_FROM_ENVIRONMENT(env);
+	DWORD dwPeriod = (DWORD)(_extensions->itPeriodMicro / 1000);
+	DWORD dwDelay = dwPeriod;
+	BOOL ret = CreateTimerQueueTimer(&_hTimer, NULL, (WAITORTIMERCALLBACK)itAlarmThunk, (PVOID)alarmThread, dwDelay, dwPeriod, 0);
+	if (!ret) {
 		/* Error creating timer */
-		timeEndPeriod(uPeriod);
+		_hTimer = NULL;
+		omrstr_printf("CreateTimerQueueTimer failed: %d", GetLastError());
```

Comment:
We can not use print here unfortunately. The tracepoint can be used however I guess Trace Engine would not be initialized at the time of call so it would not work any way. You can just remove this line or add a new tracepoint with hope that Trace Engine initialization would be fixed eventually and tracepoint would have effect